### DISCUSSION TITLE
ILI9225 with an ESP8266 - circles have missing pixels.  Weird right?

### DISCUSSION THREAD

**Discussion by klifn** [Bodmer/TFT_eSPI#2436] [Q&A - General]
I'm encountering some odd rendering when using the 2.5.22 version of TFT_eSPI with an ILI9225 display and an ESP8266.   I've included an example picture below.   Text, vertical and horizontal lines all work fine.   Non vertical lines seem to exhibit some of the same missing pixels.   A line at 45 degrees isn't drawn at all.   I have 2 identical displays and it does the same with both of them.     I just finished a project about a week ago that uses an ILI9341 display and an ESP32 and that works perfectly (it's on version 2.5.0 of TFT_eSPI).

I tried reverting back to 2.5.0 for the ILI9225 project but it behaves the same.  

I'm just using the regular circle drawing call, not the smooth version.
`  
tft.drawCircle(50,50,50,TFT_WHITE);
tft.drawCircle(75,150,35,TFT_CYAN);

`

<img width="406" alt="ili9225" src="https://user-images.githubusercontent.com/63608529/220477141-f2f8b75c-423b-41d9-bb1d-45b5884733cc.png">

Anyone have any ideas as what might be going on here?
Thanks,
Klif

**Comment by Bodmer** [comment]
It looks like the drawPixel function is not working.

I have connected an ESP8266 to an ILI9225 SPI display and it works correctly. Latest library and board package used.

Post more details of your hardware and software.

**Reply by klifn** [reply]
The hardware is an ILI9225 display and a Wemos D1 mini ESP8266.   
I'm compiling with Platform IO on an iMac running MacOS Ventura.  
I created a new Platform IO project that has nothing but the TFT_eSPI library and a small bit of code to draw the 2 circles and I still get the same behavior.    I also tried this on my MacBook Air which has a similar install of Platform IO and the results are the same.  Whatever I'm doing at least I'm doing it consistently. 😞

This isn't critical so I may try rebuilding it with an ESP32.  That seemed to work well the last time.

This is the new code.
`
#include <Arduino.h>
#include <TFT_eSPI.h>

// Use hardware SPI
TFT_eSPI tft = TFT_eSPI();

void setup() {
  // put your setup code here, to run once:
    // start the serial connection
  // Serial.begin(115200);
  Serial.begin(74880);

  // wait for serial monitor to open
  while (!Serial)
    ;

  tft.init();
  tft.setRotation(3);
  tft.fillScreen(TFT_BLACK);

  tft.drawCircle(50,50,50,TFT_WHITE);
  tft.drawCircle(75,150,35,TFT_CYAN);
  tft.drawPixel()

}

void loop() {
  // put your main code here, to run repeatedly:
  delay(1000);
}
`

And my User_Setup.h is:
`
#define USER_SETUP_INFO "User_Setup"
#define ILI9225_DRIVER
#define TFT_CS   PIN_D8  // Chip select control pin D8
#define TFT_DC   PIN_D3  // Data Command control pin
#define TFT_RST  PIN_D4  // Reset pin (could connect to NodeMCU RST, see next line)
#define TFT_MISO PIN_D6
#define TFT_MOSI PIN_D7
#define TFT_SCLK PIN_D5


#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters

// Comment out the #define below to stop the SPIFFS filing system and smooth font code being loaded
// this will save ~20kbytes of FLASH
// #define SMOOTH_FONT


#define SPI_FREQUENCY  40000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  20000000

// The XPT2046 requires a lower SPI clock rate of 2.5MHz so we define that here:
#define SPI_TOUCH_FREQUENCY  2500000
`

**Reply by klifn** [reply]
I decided to try and switch to a NodeMCU ESP8266 that I had.  I used another ILI9225 as well.   Unfortunately the results are the same.   So I tried an earlier version of the library 2.4.79, still the same.    I tried building again on my laptop for the NodeMCU and again the same result.   I seem to be pretty well infected with the dead setPixel bug.

Not sure what else to try at this point.  I think I'll give it a rest and play with my ESP32's for awhile.

**Reply by klifn** [reply]
I did try one more thing, just because this is bugging me.   I switched to the "TFT TT ILI9225" library and ran a similar test program.   It's also behaving oddly.    It will only draw filled rectangles, not even partial circles.   It too doesn't like setPixel.    I'm starting to think there is something faulty with the displays that I have.  I have 2 that behave in the same way. I bought them together from the same AliExpress vendor and perhaps there was a production issue.  Just complete speculation at this point, however I think it is fairly safe to say that it's not something I was doing wrong with the library.    I also swapped out the ILI9225 display with an ILI9341 on the original TFT_eSPI sketch, update the user_setup.h accordingly, and ran that with the ESP8266 and it of course worked fine.   Those 2 ILI9225 displays are going in the back of the drawer and I'll stick with ILI9341's.  Those seem to work great.
